libraries/AP_Vehicle/AP_Vehicle.h

```diff
@@ -201,6 +201,10 @@ class AP_Vehicle : public AP_HAL::HAL::Callbacks {
     virtual bool set_target_velaccel_NED(const Vector3f& target_vel, const Vector3f& target_accel, bool use_yaw, float yaw_deg, bool use_yaw_rate, float yaw_rate_degs, bool yaw_relative) { return false; }
     virtual bool set_target_angle_and_climbrate(float roll_deg, float pitch_deg, float yaw_deg, float climb_rate_ms, bool use_yaw_rate, float yaw_rate_degs) { return false; }
 
+    // command throttle percentage and roll, pitch, yaw target
+    // rates. For use with scripting controllers
+    virtual bool set_target_throttle_rate_rpy(float throttle_pct, float roll_rate_dps, float pitch_rate_dps, float yaw_rate_dps) { return false; }
```

Comment:
I would probably call this "set_target_rpy_rates_and_thrust()" but it's not a blocker

Reply:
It's actually used as a direct throttle demand percentage to the throttle servo, not a thrust demand that then gets converted to a throttle so I think the existing name is OK.